Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.h

```diff
@@ -106,7 +107,14 @@ namespace AzToolsFramework
             // Internal helper used by Select[|Prev|Next]ActiveComponentMode
             bool ActiveComponentModeChanged(const AZ::Uuid& previousComponentType);
 
+            struct ComponentModeItem
```

Comment:
suggestion: For `emplace_back` to work as I mentioned having a constructor for this with the two arguments as parameters would be great (plus a defaulted constructor too)